src/bson.rs

```diff
@@ -364,6 +371,9 @@ impl From<Bson> for Value {
     }
 }
 
+// 9999-12-31 23:59:59.999
+const MAX_RFC3339_TIMESTAMP: i64 = 253402300799999;
```

Comment:
Because `to_time` clamps rather than erroring (following the behavior of `to_chrono`), and `time::OffsetDateTime` maxes out at the year 9999, the conditionals checking `year` were always passing, leading to a fun off-by-one-millisecond failure in the corpus tests.  Checking against actual timestamp avoids conversion issues.